src/runtime/composables/useStrapiClient.ts

```diff
@@ -23,17 +23,24 @@ const defaultErrors = (err: FetchError) => ({
   }
 })
 
-export const useStrapiClient = () => {
+export const useStrapiClient = (options?: {
+  token?: string
+}) => {
   const nuxt = useNuxtApp()
   const baseURL = useStrapiUrl()
   const version = useStrapiVersion()
-  const token = useStrapiToken()
+  const userToken = useStrapiToken()
+  const config = useRuntimeConfig()
 
   return async <T> (url: string, fetchOptions: FetchOptions = {}): Promise<T> => {
     const headers: HeadersInit = {}
 
-    if (token && token.value) {
-      headers.Authorization = `Bearer ${token.value}`
+    if (options?.token) {
+      headers.Authorization = `Bearer ${options.token}`
+    } else if (config.strapi.defaultToken === 'user' && userToken.value) {
+      headers.Authorization = `Bearer ${userToken.value}`
+    } else if ((config.strapi.defaultToken === 'api' && config.strapi.apiToken) || config.strapi.apiToken) {
```

Comment:
Are you sure defaulting to the `config.strapi.apiToken` is the right thing to do here? For example if I set an `apiToken` to use in a specific section of my app, I might not want to use it all the time when a user is logged out.

Reply:
@benjamincanac I wan't entirely sure which way to go with it. My thinking with it this way was trying to avoid cluttering the module config with too many options considering that a token can always be passed to the composable(s).

But, I was considering maybe adding an additional boolean config (`privateApi`) or something) to allow someone to set whether to use the `apiToken` in their config as the fall back if the jwt isn't set. This might be beneficial to someone who doesn't want their API public. I'm not against that, if that works for you. It could be this instead:

```ts
    headers.Authorization = `Bearer ${options.token}`
    } else if (config.strapi.defaultToken === 'user' && userToken.value) {
      headers.Authorization = `Bearer ${userToken.value}`
    } else if ((config.strapi.defaultToken === 'api' && config.strapi.apiToken) || (config.strapi.privateApi === true && config.strapi.apiToken) {
```
I'm not sure that `privateApi` is the best name for that, but just a spur of the moment idea.

Reply:
I'm wondering, why would you use an api token in your app instead of just using a public endpoint? As this http call will be client-side, all users will be able to see this token so it comes to the same thing.

Reply:
Hmm, yeah. Potentially, a bit of a misconception on my part. I'm sure you would know better than I. Most of my understanding is just from some articles rather than a wealth of experience. Such as:

https://cloud.google.com/endpoints/docs/openapi/when-why-api-key#:~:text=API%20keys%20provide%20project%20authorization,-To%20decide%20which&text=By%20identifying%20the%20calling%20project,or%20enabled%20in%20the%20API.

Reply:
Sorry it took so long to catch up, but the only use-case I see to use api tokens in nuxt is if it's used server-side as these tokens are private. However, as mentioned here https://github.com/nuxt-modules/strapi/issues/320#issuecomment-1454644265, the module does not export server functions as of right now.